lib/Matrix.js

```diff
@@ -202,7 +184,32 @@ function useMatrixProvider(auth) {
                 setRooms(setRoom(room.roomId));
                 setRoomContents(setMessages(room.roomId));
             }
-        }, [directMessages, knockingMembers, setDirectMessages, setKnockingMembers, setMessages, setRoom, setRoomContents, setRooms, setSpaces],
+
+            // Update our internal store for pending knocks if this timeline event contained info about a knock
+            if (
+                event.getType() === EventType.RoomMember &&
+                room.getMember(matrixClient.getUserId()).powerLevel >= room.getLiveTimeline().getState(EventTimeline.FORWARDS).getStateEvents(EventType.RoomPowerLevels, '')?.getContent()['invite'] &&
+                event.getContent().membership === 'knock'
+            ) {
+                // Looks like we got a new knock for a room that we can manage them for ...
+                setKnockingMembers(map => {
+                    map.set(`${room.roomId}|${event.target.userId}`, {
+                        roomId: room.roomId,
+                        userId: event.target.userId,
+                        reason: event.getContent()?.reason,
+                    });
+                });
+            } else if (
+                event.getType() === EventType.RoomMember &&
+                knockingMembers.has(`${room.roomId}|${event.target.userId}`) &&
+                event.getContent().membership !== 'knock'
+            ) {
+                // Looks like some previously existing knock was handled (or retracted?) ...
+                setKnockingMembers(map => {
+                    map.delete(`${room.roomId}|${event.target.userId}`);
+                });
+            }
+        }, [directMessages, knockingMembers, matrixClient, setDirectMessages, setKnockingMembers, setMessages, setRoom, setRoomContents, setRooms, setSpaces],
```

Comment:
I think it would work if we wrapped the entire knock logic here in a condition 
```javascript
 if (matrixClient.isInitialSyncComplete()) {
// Update our internal store for pending knocks if this timeline event contained info about a knock
...
}
```

Then, `initializePendingKnocks` can handle the sync and this logic can handle new events.